File: src/main/java/com/pairingplanet/pairing_planet/dto/user/MyProfileResponseDto.java
package com.pairingplanet.pairing_planet.dto.user;

import lombok.Builder;

@Builder
public record MyProfileResponseDto(
        UserDto user,
        long recipeCount, // 내가 만든 레시피 수
        long logCount     // 내가 남긴 로그 수
) {}

File: lib/data/models/user/my_profile_response_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'user_dto.dart';

part 'my_profile_response_dto.g.dart';

@JsonSerializable()
class MyProfileResponseDto {
  final UserDto user;
  final int recipeCount;
  final int logCount;
  final int savedCount;

  MyProfileResponseDto({
    required this.user,
    required this.recipeCount,
    required this.logCount,
    required this.savedCount,
  });

  factory MyProfileResponseDto.fromJson(Map<String, dynamic> json) =>
      _$MyProfileResponseDtoFromJson(json);
  Map<String, dynamic> toJson() => _$MyProfileResponseDtoToJson(this);
}
